samza-test/src/test/java/org/apache/samza/test/processor/TestZkLocalApplicationRunner.java

```diff
@@ -0,0 +1,498 @@
+          return result;
+        };
+        MessageStream<String> inputStream = graph.getInputStream(inputTopic, messageBuilder);
+        OutputStream<String, String, String> outputStream = graph.getOutputStream(outputTopic, event -> null, event -> event);
+        inputStream.sendTo(outputStream);
+      } catch (Exception e) {
+        if (appLifeCycleListener != null) {
+          appLifeCycleListener.onExceptionOccurred(e);
+        }
+      }
+    }
+  }
+
+  @Override
+  public void setUp() {
+    super.setUp();
+    testStreamAppName = String.format("test-app-name-%s", UUID.randomUUID().toString());
+    testStreamAppId = String.format("test-app-id-%s", UUID.randomUUID().toString());
+    inputKafkaTopic = String.format("test-input-topic-%s", UUID.randomUUID().toString());
+    outputKafkaTopic = String.format("test-output-topic-%s", UUID.randomUUID().toString());
+    for (String kafkaTopic : ImmutableList.of(inputKafkaTopic, outputKafkaTopic)) {
+      LOGGER.info("Creating kafka topic: {}.", kafkaTopic);
+      TestUtils.createTopic(zkUtils(), kafkaTopic, 5, 1, servers(), new Properties());
+    }
+  }
+
+  @Override
+  public void tearDown() {
+    for (String kafkaTopic : ImmutableList.of(inputKafkaTopic, outputKafkaTopic)) {
+      LOGGER.info("Deleting kafka topic: {}.", kafkaTopic);
+      AdminUtils.deleteTopic(zkUtils(), kafkaTopic);
+    }
+    super.tearDown();
+  }
+
+  // Helper methods that are shared by tests.
+  private List<String> readProcessorIdsFromZK(String zkPathPrefix, List<String> processorIds) {
+    ZkClient zkClient = new ZkClient(zkConnect());
+    ZkKeyBuilder zkKeyBuilder = new ZkKeyBuilder(zkPathPrefix);
+    ZkUtils zkUtils = new ZkUtils(zkKeyBuilder, zkClient, 10000);
+    return zkUtils.getActiveProcessorsIDs(processorIds);
+  }
+
+  private String readJobModelVersionFromZK(String basePath) {
+    ZkUtils zkUtils = null;
+    try {
+      zkUtils = getZKUtils(basePath);
+      return zkUtils.getJobModelVersion();
+    } finally {
+      if (zkUtils != null) {
+        zkUtils.close();
+      }
+    }
+  }
+
+  private JobModel readJobModelFromZK(String basePath) {
+    ZkUtils zkUtils = null;
+    try {
+      zkUtils = getZKUtils(basePath);
+      zkUtils.connect();
+      return zkUtils.getJobModel(zkUtils.getJobModelVersion());
+    } finally {
+      if (zkUtils != null) {
+        zkUtils.close();
+      }
+    }
+  }
+
+  private ZkUtils getZKUtils(String basePath) {
+    ZkClient zkClient = new ZkClient(zkConnect());
+    ZkKeyBuilder zkKeyBuilder = new ZkKeyBuilder(basePath);
+    return new ZkUtils(zkKeyBuilder, zkClient, 10000);
+  }
+
+  private void publishKafkaEvents(String topic, int numMessages, String streamProcessorId) {
+    KafkaProducer producer = getKafkaProducer();
+    for (int messageIndex = 0; messageIndex < numMessages; messageIndex++) {
+      try {
+        LOGGER.info("Publish kafka event with index : {} for stream processor: {}.", messageIndex, streamProcessorId);
```

Comment:
Prefer info, I would want all lines to print during one run.  Helps debugging. Don't have strong preference.